boa/src/vm/mod.rs

```diff
@@ -45,12 +64,16 @@ impl<'a> VM<'a> {
 
     pub fn run(&mut self) -> Result<Value> {
         let _timer = BoaProfiler::global().start_event("runVM", "vm");
-        let mut idx = 0;
+        self.idx = 0;
+
+        while self.idx < self.instructions.len() {
+            if self.is_trace {
+                self.trace_print(false);
+            };
 
-        while idx < self.instructions.len() {
             let _timer =
-                BoaProfiler::global().start_event(&self.instructions[idx].to_string(), "vm");
-            match self.instructions[idx] {
+                BoaProfiler::global().start_event(&self.instructions[self.idx].to_string(), "vm");
+            match self.instructions[self.idx] {
```

Comment:
Here I see we are iterating and using the index to retrieve the instruction. Couldn't we do this with iterators?

Reply:
I’m going to merge and come back to this